src/kv/value.rs

```diff
@@ -83,6 +83,11 @@ macro_rules! as_sval {
 /// - Using the `ToValue` trait.
 /// - Using the standard `From` trait.
 ///
+// REVIEW: maybe add a section about which conversion method to use and when? Or
```

Comment:
Good call :+1:

The preference would be for `capture_*`, then `from_*`. The reason being if you `Value::capture_display(&42)` you'll get a number, but if you use `Value::from_display(&42)` you'll get a string. I'd be open to some different naming that made this more obvious. We may consider also only adding them for `Display` and `Debug` since `serde` and `sval` are naturally structured. We could also consider offering just a single `capture` method that is what `capture_display` currently is, and using that by default in the macros. The goal is that by default if you log a boolean or number you should get a boolean or number, and you shouldn't _have_ to implement any non-standard-library traits like `ToValue` to log something.

I'm open to suggestions though. What do you think?

Reply:
> Does it make sense to change the macros to always call log::ToValue::to_value($value)?

The idea was to try discourage libraries from implementing `ToValue` themselves so it doesn't become yet another widespread piece of trivia that library authors need to keep on top of. Defaulting to `Display + 'static` means a lot of library types like `Uuid` can be captured, but the drawback of throwing lifetime errors at you if you try log a non-`'static` value would be unpleasant.

For what it's worth, I would be ok with using `ToValue` by default in addition to adding some easy syntax for `Display`/`Debug`/`Serialize`/`Value`. If we did that then we could remove the `capture_*` methods altogether; they only exist to support downcasting to tell if we're logging numbers/booleans etc.

> Then, once specialisation is stablised

🥲